### PR TITLE
Added tunnel arg to jenkins::slave, to support this arg for the swarm…

### PR DESCRIPTION
**Pull request by fipp**
Support for the tunnel argument added to the Jenkins Swarm Plugin in version 1.23.

### CODE REVIEW

File: manifests/slave.pp [hunk 1]
```diff
@@ -77,6 +80,7 @@
   $slave_name               = undef,
   $description              = undef,
   $masterurl                = undef,
+  $tunnel                   = undef,
```
Comment (jhoblitt) [inline manifests/slave.pp]: Please validate the input (I realize this module is inconsistent about input validation but we should do it for all new params).

File: templates/jenkins-slave-defaults.erb [hunk 2]
```diff
@@ -92,4 +97,4 @@ if [ -n "$AUTO_DISCOVERY_ADDRESS" ]; then
   AUTO_DISCOVERY_ADDRESS_ARG="-autoDiscoveryAddress '$AUTO_DISCOVERY_ADDRESS'"
 fi
 
-JENKINS_SLAVE_ARGS="-mode $JENKINS_SLAVE_MODE -executors $EXECUTORS $CREDENTIALS_ARG $NAME_ARG $MASTER_URL_ARG $LABELS_ARG $FSROOT_ARG $AUTO_DISCOVERY_ADDRESS_ARG $TOOL_LOCATIONS_ARG $DESCRIPTION_ARG $OTHER_ARGS"
+JENKINS_SLAVE_ARGS="-mode $JENKINS_SLAVE_MODE -executors $EXECUTORS $CREDENTIALS_ARG $NAME_ARG $MASTER_URL_ARG $TUNNEL_ARG $LABELS_ARG $FSROOT_ARG $AUTO_DISCOVERY_ADDRESS_ARG $TOOL_LOCATIONS_ARG $DESCRIPTION_ARG $OTHER_ARGS"
```
Comment (jhoblitt) [inline templates/jenkins-slave-defaults.erb]: This needs rspec coverage.

**Comment by rtyler** [comment]
*bump* on @jhoblitt's feedback

**Comment by AndreaGiardini** [comment]
Can this be merged plese?

**Comment by vStone** [comment]
Picked up in #840

**Pull request closed without merging**